src/UICrafter.Core/Components/PocButtonNoMistakes.razor

```diff
@@ -0,0 +1,60 @@
+@using MudBlazor
+@using UICrafter.Core.Models
+@using System.Diagnostics
+@using Microsoft.AspNetCore.Components
+@using System.Net.Http
+@using System.Net.Http.Json
+@using UICrafter.Core.Utility
+
+@inject HttpClient Http
+
+@using UICrafter.Core.UIComponents
+
+<MudButton OnClick="CallApi"
+           Color="@buttonColor"
+           Disabled="@isDisabled">
+    @Button.Button.Label
+</MudButton>
+
+@code {
+    [Parameter]
+    public required UIComponent Button { get; set; }
+
+    [CascadingParameter]
+    public required StupidStore StupidStore { get; set; }
+
+    private bool isDisabled = false;
+    private Color buttonColor = Color.Primary;
+
+    private async Task CallApi()
+    {
+        try
+        {
+            isDisabled = true;
+            var originalLabel = Button.Button.Label;
+            Button.Button.Label = "Loading...";
```

Comment:
I am not fan of such a loading thing. it is better to replace the text with a loading indicator or something along those lines. Keep it for now but should be reworked.